swig/StringArray.hpp

```diff
@@ -1,13 +1,16 @@
 /*!
  * Copyright (c) 2020 Microsoft Corporation. All rights reserved.
  * Licensed under the MIT License. See LICENSE file in the project root for license information.
+ *
+ * Author: Alberto Ferreira
  */
-#ifndef __STRING_ARRAY_H__
-#define __STRING_ARRAY_H__
+#ifndef LIGHTGBM_SWIG_STRING_ARRAY_H_
+#define LIGHTGBM_SWIG_STRING_ARRAY_H_
```

Comment:
To match the filename.
```suggestion
#ifndef LIGHTGBM_SWIG_STRINGARRAY_HPP_
#define LIGHTGBM_SWIG_STRINGARRAY_HPP_
```
@shiyu1994 Are you OK to leave this SWIG-only-related file named in camel case way? I guess renaming will be a serious breaking change for MMLSpark, for example.

Reply:
I can rename StringArray->string_array, won't change anything in MMLSpark, but that will trigger a huge git diff as we already fixed its cpp lint issues. Hence we should do this on a separate PR, :+1: @StrikerRUS ?

Reply:
> I can rename StringArray->string_array, won't change anything in MMLSpark,

Nice to hear!

> Hence we should do this on a separate PR, 👍 @StrikerRUS ?

Yes, please!